src/views/List.tsx

```diff
@@ -1,21 +1,35 @@
-import React from 'react';
-import { ListItem as ListItemComponent } from '../components/ListItem';
-import { ListItem } from '../api';
+import { useState } from "react";
+import { ListItem as ListItemComponent } from "../components/ListItem";
+import { FilterList as FilterListComponent } from "../components/FilterList";
+import { ListItem } from "../api";
 
 interface Props {
 	data: ListItem[];
 }
 
 export function List({ data }: Props) {
-	const hasItem = data.length !== 0;
+	const [searchTerm, setSearchTerm] = useState<string>("");
+
+	const filteredData = data.filter((item) =>
+		item.name.toLowerCase().includes(searchTerm.toLowerCase()),
+	);
+
+	const hasItem = filteredData.length !== 0;
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
+
+			<FilterListComponent
+				searchTerm={searchTerm}
+				setSearchTerm={setSearchTerm}
+			/>
+
 			<ul>
 				{hasItem &&
-					data.map((item) => (
+					filteredData.map((item) => (
```

Comment:
Praise: All of this functionality looks clean!
Nitpick: Why not call this variable filteredListData?